File: apps/api/internal/subscription/dto/subscription.go
package dto

import (
	"time"

	_jetModel "github.com/cloudy-clip/api/internal/common/database/.jet/model"
	"github.com/cloudy-clip/api/internal/subscription/model"
)

type Subscription struct {
	Plan               *Plan                                 `json:"plan"`
	CanceledAt         *time.Time                            `json:"canceledAt"` // Nullable
	CancellationReason *model.SubscriptionCancellationReason `json:"cancellationReason"`
}

func NewSubscription(subscriptionModel _jetModel.Subscription, planDto *Plan) *Subscription {
	return &Subscription{
		Plan:               planDto,
		CanceledAt:         subscriptionModel.CanceledAt,
		CancellationReason: subscriptionModel.CancellationReason,
	}
}

File: packages/web-ui-common/src/entitlement/models/subscription.ts
import { Plan } from './plan';

export type Subscription = {
  canceledAt: string | null;
  cancellationReason: CancellationReason | null;
  plan: Plan;
};

export type CancellationReason = 'REQUESTED_BY_USER' | 'PAYMENT_FAILED' | 'PAYMENT_DISPUTED';
